### ISSUE TITLE
time.Time gets mapped as a nested struct in version 0.7 and later

### ISSUE THREAD

**Issue by tbflw** [jonbodner/proteus#27]
After the support for nested structs in 0.7 (which is great!), fields of type `time.Time` are now incorrectly mapped as nested struct types since `time.Time` is a struct but it does not implement `sql.Scanner`. This causes all our `time.Time` fields not to be mapped correctly to our postgres db where we want to map directly to columns of type `TIMESTAMP WITH TIMEZONE` .  

The culprit is the recurse in line 97 in `mapper/mapper.go` inside function `buildColFieldMap`:
```
			//if this is a struct, recurse
			// we are going to use an prof value as a prefix
			// then we go through each of the fields in the struct
			// if any of them aren't structs and have prof tags, we
			// prepend the parent struct and store off a fieldi
			// only if this doesn't implement a scanner. If it does, then go with the scanner
			if sf.Type.Kind() == reflect.Struct && !reflect.PtrTo(sf.Type).Implements(scannerType) {
				buildColFieldMap(sf.Type, childFieldInfo, colFieldMap)
			} else {
				colFieldMap[strings.SplitN(tagVal, ",", 2)[0]] = childFieldInfo
			}
```
I know mapping of `time.Time` is very driver specific - maybe add a special case handling this?

**Comment by jonbodner** [comment]
oooh, yeah, I'll put in a special case for time.Time. It's one of the special cases for driver.Valuer:

```
int64
float64
bool
[]byte
string
time.Time
```

**Comment by jonbodner** [comment]
fixed in v0.8.1
